src/ngraph/runtime/gpu/gpu_kernel_emitters.cpp

```diff
@@ -45,6 +45,129 @@ void runtime::gpu::kernel::emit_memcpyDtD(codegen::CodeWriter& writer,
     return;
 }
 
+void runtime::gpu::kernel::emit_cudnnConvolutionDescriptor(codegen::CodeWriter& writer,
+                                                           const std::string& name,
+                                                           const CoordinateDiff& padding,
+                                                           const Strides& window_movement_strides,
+                                                           const Strides& window_dilation_strides,
+                                                           const std::string& mode,
+                                                           const std::string& data_type)
+{
+    writer << "cudnnConvolutionDescriptor_t " << name << ";\n";
+    writer << "cudnnCreateConvolutionDescriptor(&" << name << ");\n";
+
+    if (padding.size() == 2)
+    {
+        writer << "cudnnSetConvolution2dDescriptor(" << name << ", " << padding[0] << ", "
+               << padding[1] << ", " << window_movement_strides[0] << ", "
+               << window_movement_strides[1] << ", " << window_dilation_strides[0] << ", "
+               << window_dilation_strides[1] << ", " << mode << ", " << data_type << ");\n";
+    }
+    else
+    {
+        writer << "const int " << name << "_padding[] = {" << join(padding) << "};\n";
+        writer << "const int " << name << "_window_movement_strides[] = {"
+               << join(window_movement_strides) << "};\n";
+        writer << "const int " << name << "_window_dilation_strides[] = {"
+               << join(window_dilation_strides) << "};\n";
+
+        writer << "cudnnSetConvolutionNdDescriptor(" << name << ", " << padding.size() << ", "
+               << name << "_padding, " << name << "_window_movement_strides, " << name
+               << "_window_dilation_strides, " << mode << ", " << data_type << ");\n";
+    }
+}
+
+void runtime::gpu::kernel::emit_cudnnFilterDescriptor(codegen::CodeWriter& writer,
+                                                      const std::string& name,
+                                                      const std::string& format,
+                                                      const std::string& data_type,
+                                                      const Shape& shape)
+{
+    std::vector<size_t> dimensions;
```

Comment:
nitpick: this logic seems to be duplicated across `emit_cudnnFilterDescriptor` and `emit_cudnnTensorDescriptor`. Can we refactor it into a small helper?